docs/building-apps/wallet/sep30.mdx

```diff
@@ -0,0 +1,301 @@
+---
+title: Recovery
+sidebar_position: 60
+---
+
+import Header from "./component/header.mdx";
+import { WalletCodeExample as CodeExample } from "@site/src/components/WalletCodeExample";
+
+<Header WIPLangs={["dart"]} />
+
+The [Sep-30] standard defines the standard way for an individual (e.g., a user or wallet) to regain access to their Stellar account after losing its private key without providing any third party control of the account. During this flow the wallet communicates with one or more recovery signer servers to register the wallet for a later recovery if it's needed.
+
+## Create Recoverable Account
+
+First, let's create an account key, a device key, and a recovery key that will be attached to the account.
+
+<CodeExample>
+
+```typescript
+const accountKp = wallet.stellar().account().createKeypair();
+const deviceKp = wallet.stellar().account().createKeypair();
+const recoveryKp = wallet.stellar().account().createKeypair();
+```
+
+```kotlin
+  val accountKp = wallet.stellar().account().createKeyPair()
+  val deviceKp = wallet.stellar().account().createKeyPair()
+  val recoveryKp = wallet.stellar().account().createKeyPair()
+```
+
+</CodeExample>
+
+The `accountKp` is the wallet's main account. The `deviceKp` we will be adding to the wallet as a signer so a device (eg. a mobile device a wallet is hosted on) can take control of the account. And the `recoveryKp` will be used to identify the key with the recovery servers.
+
+Next, let's identify the recovery servers and create our recovery object:
+
+<CodeExample>
+
+```typescript
+const server1Key = "server1";
+const server1 = {
+  endpoint: "recovery-example.com",
+  authEndpoint: "auth-example.com",
+  homeDomain: "test-domain",
+};
+
+const server2Key = "server2";
+const server2 = {
+  endpoint: "recovery-example2.com",
+  authEndpoint: "auth-example2.com",
+  homeDomain: "test-domain2",
+};
+
+const recovery = wallet.recovery({
+  servers: { [server1Key]: server1, [server2Key]: server2 },
+});
+```
+
+```kotlin
+val first = RecoveryServerKey("first")
+val second = RecoveryServerKey("second")
+val firstServer = RecoveryServer("recovery.example.com", "auth.example.com", "example.com")
+val secondServer = RecoveryServer("recovery2.example.com", "auth2.example.com", "example.com")
+val servers = mapOf(first to firstServer, second to secondServer)
+val recovery = wallet.recovery(servers)
+```
+
+</CodeExample>
+
+Next, we need to define SEP-30 identities. In this example we are going to use the same identity on both recovery servers.
```

Comment:
Do you mind adding one more identity for the second server, using email? (just for clarity and because using same identity on both servers is an anti-pattern)

Reply:
like this? https://github.com/stellar/stellar-docs/pull/261/commits/85e99584b6df68d353af0750e400b391f6a8f470

when I tried setting the identity with just email like this:
```
const identity2 = {
  role: RecoveryRole.OWNER,
  authMethods: [
    {
      type: RecoveryType.EMAIL,
      value: "my-email@example.com",
    },
  ],
};
```

I was getting a 404 with the recovery server. From my understanding the recovery server identifies the account from the recoveryKey in the authToken right? Do you know how it should work when the identity is email only?